auth/src/main/java/software/amazon/awssdk/auth/signer/params/AwsS3V4SignerParams.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.auth.signer.params;
+
+public class AwsS3V4SignerParams extends Aws4SignerParams {
+
+    private final Boolean enableChunkedEncoding;
+    private final Boolean enablePayloadSigning;
+
+    private AwsS3V4SignerParams(Builder builder) {
+        super(builder);
+        this.enableChunkedEncoding = builder.enableChunkedEncoding;
+        this.enablePayloadSigning = builder.enablePayloadSigning;
+    }
+
+    public Boolean enableChunkedEncoding() {
+        return enableChunkedEncoding;
+    }
+
+    public Boolean enablePayloadSigning() {
+        return enablePayloadSigning;
+    }
+
+    public static Builder builder() {
+        return new Builder();
+    }
+
+    // Default values from v1
+    public static final class Builder extends Aws4SignerParams.Builder<Builder> {
+        public static final boolean DEFAULT_CHUNKED_ENCODING_ENABLED = true;
+        public static final boolean DEFAULT_PAYLOAD_SIGNING_ENABLED = false;
+
+        private Boolean enableChunkedEncoding = DEFAULT_CHUNKED_ENCODING_ENABLED;
+        private Boolean enablePayloadSigning = DEFAULT_PAYLOAD_SIGNING_ENABLED;
```

Comment:
Do we want to disable payload signing for all operations by default? It appears that's what we do for V1 but I wonder if we should have limited it to PutObject and UploadPart. Can you ask the team or S3 what the ideal behavior is?

Reply:
Right now, I am using the v1 defaults. I will follow up with other SDKs about their behavior